Graylog2/graylog2-server
# Use GrokPatternRegistry to reload saved grok patterns

## Description
Prior to this change, a edit on a grok pattern did not
reflect on the used pattern in GrokExtractors

This change uses the GrokPatternRegistry which is
contains all grok patterns and is connected to the
cluster event bus and reloads when a grok pattern is
edit.

One test needed to be moved since it was testing the
grok pattern service which is not used anymore in the
GrokExtractor directly. There for the test was moved
to the MongoDbGrokPatternServiceTest where it belongs.

Fixes #5833

## How Has This Been Tested?
- Create a GrokExtractor with a GrokPattern
- Edit GrokPattern and send a message to the input
- Check that change on GrokPattern reflects the extractors work

## Types of changes

## Checklist:
